### ISSUE TITLE
No way to change top navigation root level colors

### ISSUE THREAD

**Issue by PingusPepan** [SnowdogApps/magento2-theme-blank-sass#92]
I found no other way to change top navigation items color than to edit vendor/magento-ui/_navigation.scss file. I guess there should be a way to change the colors via _variables.scss. I used the `$navigation-desktop-level0-item__color` for desktop root items and added `$submenu-desktop-item__color` for submenu items.
If there is a way but I was just unable to find it, let me know!

**Comment by PingusPepan** [comment]
There were more things I wasn't able to change for the navigation. I just used the variable names from less library and seems to be working for me. Is there any reason as to why it is set up this way for sass or is it just waiting to be completed?

**Comment by Igloczek** [comment]
SASS variables works like in any other programing language - they have execution order and scopes. In LESS you can modify any variable in any place and it will affect whole codebase - this sucks when you care about high maintainability of codebase.

SASS version of Magento UI have defined default mixins parameters values and you have to overwrite them if you need some customization.

So if you want to change value of `$_submenu-item-color` parameter of `lib-main-navigation-desktop` mixin, you have to pass this parameter on place were mixin is invoked.
In this case it will be `[...]/theme-blank-sass/styles/blocks/_navigation.scss` line 296.

``` sass
@include lib-main-navigation-desktop(
    $_submenu-item-color: $your-variable-with-color-of-navigation-links // or just value
);
```

**Comment by PingusPepan** [comment]
Thanks for the mixin way, I figured it would work this way but I looked for solution where I wouldn't have to overwrite whole _navigation.scss file. Thus trying to do this with variables.

Take `$navigation-desktop-level0-item__color` for example:
For `@mixin lib-main-navigation-desktop` the corresponding variable should be `$_nav-level0-item-color`.
`$_nav-level0-item-color` is instead set to `$color-gray34` (`styles/vendor/magento-ui/_navigation.scss:247`)
`$navigation-desktop-level0-item__color` is used for `$_submenu-item-color` (`styles/vendor/magento-ui/_navigation.scss:279`). I couldn't find a reason for this.

`lib/web/css/source/lib/_navigation.less`:
`@_nav-level0-item-color: @navigation-desktop-level0-item__color`
`@_submenu-item-color: @submenu-desktop-item__color`
vs
`styles/vendor/magento-ui/_navigation.scss`:
`$_nav-level0-item-color : $color-gray34`
`$_submenu-item-color : $navigation-desktop-level0-item__color`

If changing the submenu item color by changing the variable `$navigation-desktop-level0-item__color` in _variables.scss works, why shouldn't it work like this for other variables?
At least change the working variable names to correspond with Magento less variables. Rename `$navigation-desktop-level0-item__color` to `$_submenu-item-color` so there is less confusion.

**Comment by Igloczek** [comment]
I was simplifying inheritance of variables, replacing them with "source" variables, b/c they was often nested on few levels ( color -> color-in-block -> color-in-block-element -> color-in-block-element-state -> color-in-element-state-child-element ... #mess).

Goal is to make modifications by invoking mixins with custom parameters (so you have to overwrite `_naviagation.scss` in this case), instead of editing global variables. Ability to change color via `$navigation-desktop-level0-item__color` is a bug and will be fixed soon.

**Comment by PingusPepan** [comment]
Okay, thanks for the explanation. I will overwrite the file and pass the variables to the mixin then.

**Comment by Igloczek** [comment]
Fixed in 0.8.1.
